src/core/stream/stream_loader.ts

```diff
@@ -169,18 +175,38 @@ export default function StreamLoader({
       sourceBufferManager,
       segmentPipelinesManager,
       bufferOptions
-    ).pipe(mergeMap((evt) : Observable<IStreamLoaderEvent> => {
-      switch (evt.type) {
-        case "end-of-stream":
-          return maintainEndOfStream(mediaSource)
-            .pipe(ignoreElements(), takeUntil(cancelEndOfStream$));
-        case "resume-stream":
-          cancelEndOfStream$.next(null);
-          return EMPTY;
-        default:
-          return onBufferEvent(evt);
-      }
-    }));
+    ).pipe(
+      mergeMap((evt) : Observable<IStreamLoaderEvent> => {
+        switch (evt.type) {
+          case "end-of-stream":
+            return maintainEndOfStream(mediaSource)
+              .pipe(ignoreElements(), takeUntil(cancelEndOfStream$));
+          case "resume-stream":
+            cancelEndOfStream$.next(null);
+            return EMPTY;
+          case "discontinuity-encountered":
+            if (SourceBufferManager.isNative(evt.value.bufferType)) {
+              log.warn("explicit discontinuity seek", evt.value.nextTime);
+              mediaElement.currentTime = evt.value.nextTime;
+            }
+            return EMPTY;
+          case "needs-manifest-refresh":
+            log.debug("needs manifest to be refreshed");
```

Comment:
I prepend now a `Stream:` namespace in the StreamLoader's logs, to better help us debugging with logs on